Source/WebCore/page/Quirks.cpp

```diff
@@ -1885,6 +1886,13 @@ bool Quirks::needsDesktopUserAgent(const URL& url)
     return needsDesktopUserAgentInternal(url);
 }
 
+bool Quirks::needsPartitionedCookies(const ResourceRequest& request)
+{
+    if (request.isTopSite())
+        return false;
+    return request.url().protocolIs("https"_s) && request.url().host().endsWith(".billpaysite.com"_s);
```

Comment:
Does this need to be limited to `https`? I think we generally use `URL::protocolIsInHTTPFamily()` for this (which just looks up a cached bit).

Reply:
I suppose not. Using a bill payment site with http would make me uncomfortable, but you're right that this quirk doesn't need to be limited to https.